pkg/node/address.go

```diff
@@ -574,14 +569,18 @@ func GetIngressIPv6() net.IP {
 	return getLocalNode().IPv6IngressIP
 }
 
-// GetEncryptKeyIndex returns the encryption key value for the local node.
-// With IPSec encryption, this is equivalent to GetIPsecKeyIdentity().
-// With WireGuard encryption, this function returns a non-zero static value
-// if the local node has WireGuard enabled.
-func GetEncryptKeyIndex() uint8 {
+// GetEndpointEncryptKeyIndex returns the encryption key value for an endpoint
+// owned by the local node.
+// With IPSec encryption, this is the ID of the currently loaded key.
+// With WireGuard, this returns a non-zero static value if the local node
+// has announced a WireGuard public key.
+// Note that the key index returned by this function is only valid for _endpoints_
+// of the local node. If you want to obtain the key index of the local node itself,
+// access the `EncryptionKey` field via the LocalNodeStore.
+func GetEndpointEncryptKeyIndex() uint8 {
 	switch {
 	case option.Config.EnableIPSec:
-		return GetIPsecKeyIdentity()
+		return getLocalNode().EncryptionKey
 	case option.Config.EnableWireguard:
 		if len(GetWireguardPubKey()) > 0 {
```

Comment:
I guess we could additionally simplify this logic by dropping the `GetWireguardPubKey` check, given that the key is always set if wireguard is enabled (unless an error occurred, but in that case we should have already bailed out).

Reply:
Agreed!